ballista/rust/executor/src/executor_server.rs

```diff
@@ -368,16 +425,31 @@ impl<T: 'static + AsLogicalPlan, U: 'static + AsExecutionPlan> TaskRunnerPool<T,
 
         //2. loop for task fetching and running
         let executor_server = self.executor_server.clone();
+        let mut task_runner_shutdown = graceful.subscribe_for_shutdown();
+        // Not used directly.
+        let task_runner_complete = graceful.shutdown_complete_tx.clone();
         tokio::spawn(async move {
             info!("Starting the task runner pool");
+
+            // Drop and notifies the receiver half that the shutdown is complete
```

Comment:
```suggestion
            // Capture `task_runner_complete`. This will be dropped when this background task returns, indicating that shutdown is complete
```

Reply:
I will take @avantgardnerio suggestion and call the drop explicitly.